src/Cli.Tests/UtilsTests.cs

```diff
@@ -1,324 +1,325 @@
+        string? issuer,
+        bool expectSuccess)
+    {
+        Assert.AreEqual(
+            expectSuccess,
+            ValidateAudienceAndIssuerForJwtProvider(authenticationProvider, audience, issuer)
+        );
+    }
 
-        /// <summary>
-        /// Test to verify that CLI is able to figure out if the api path prefix for rest/graphql contains invalid characters.
-        /// </summary>
-        [DataTestMethod]
-        [DataRow("/", ApiType.REST, true, DisplayName = "Only forward slash as api path")]
-        [DataRow("/$%^", ApiType.REST, false, DisplayName = "Api path containing only reserved characters.")]
-        [DataRow("/rest-api", ApiType.REST, true, DisplayName = "Valid api path")]
-        [DataRow("/graphql@api", ApiType.GraphQL, false, DisplayName = "Api path containing some reserved characters.")]
-        [DataRow("/api path", ApiType.REST, true, DisplayName = "Api path containing space.")]
-        public void TestApiPathIsWellFormed(string apiPath, ApiType apiType, bool expectSuccess)
-        {
-            Assert.AreEqual(expectSuccess, IsApiPathValid(apiPath, apiType));
-        }
+    /// <summary>
+    /// Test to verify that when DAB_ENVIRONMENT variable is set, also base config and
+    /// dab-config.{DAB_ENVIRONMENT}.json file is present, then when DAB engine is started, it will merge
+    /// the two config and use the merged config to startup the engine.
+    /// Here, baseConfig(dab-config.json) has no connection_string, while dab-config.Test.json has a defined connection string.
+    /// once the `dab start` is executed the merge happens and the merged file contains the connection string from the
+    /// Test config.
+    /// Scenarios Covered:
+    /// 1. Merging of Array: Complete override of Book permissions from the second config (environment based config).
+    /// 2. Merging Property when present in both config: Connection string in the second config overrides that of the first.
+    /// 3. Non-merging when a property in the environmentConfig file is null: {data-source.options} is null in the environment config,
+    /// So it is added to the merged config as it is with no change.
+    /// 4. Merging when a property is only present in the environmentConfig file: Publisher entity is present only in environment config,
+    /// So it is directly added to the merged config.
+    /// 5. Properties of same name but different level do not conflict: source is both a entityName and a property inside book entity, both are
+    /// treated differently.
+    /// </summary>
+    [TestMethod]
+    public void TestMergeConfig()
+    {
+        MockFileSystem fileSystem = new();
+        fileSystem.AddFile(RuntimeConfigLoader.DEFAULT_CONFIG_FILE_NAME, new MockFileData(BASE_CONFIG));
+        fileSystem.AddFile("dab-config.Test.json", new MockFileData(ENV_BASED_CONFIG));
+
+        RuntimeConfigLoader loader = new(fileSystem);
+        bool old = RuntimeConfigLoader.CheckPrecedenceForConfigInEngine;
```

Comment:
Is this property being toggled between true/false because other tests depend on the default value true?

Reply:
Since this is a static it's good to preserve state when we know we're going to override it in a test and return it to what it used to be.

Could we get away without it? Probably, but it's safer this way (until we can get away from shared state 😉)